Rest/Content/ContentType.php

```diff
@@ -0,0 +1,379 @@
+        } else {
+            $languages = $this->configResolver->getParameter('languages');
+        }
+
+        if (empty($languages)) {
+            throw new LogicException(sprintf('No languages found using siteAccess or mandatorId'));
+        }
+
+        return $languages;
+    }
+
+    /**
+     * Generate export files.
+     *
+     * @param $languages
+     * @param $chunkDir
+     * @param array $options
+     *
+     * @return array
+     */
+    private function generateFiles($languages, $chunkDir, array $options)
+    {
+        $urls = array();
+
+        foreach ($options['contentTypeIds'] as $contentTypeId) {
+            $count = $this->countContentByContentTypeId($contentTypeId, $options);
+
+            $this->logger->info(sprintf('fetching %s items of contentTypeId %s', $count, $contentTypeId));
+
+            foreach ($languages as $lang) {
+                $options['lang'] = $lang;
+
+                for ($i = 1; $i <= ceil($count / $options['pageSize']); ++$i) {
+                    $filename = sprintf('%d_%s_%d', $contentTypeId, $lang, $i);
+                    $options['page'] = $i;
+
+                    $content = $this->prepareContent(
+                        $this->getContentItems($options, $contentTypeId),
+                        $options
+                    );
+
+                    $chunkPath = $options['documentRoot'] . '/var/export' . $chunkDir . $filename;
+
+                    $this->generateFile($content, $chunkPath, $options);
+
+                    $url = sprintf(
+                        '%s/api/ezp/v2/ez_recommendation/v1/exportDownload%s%s',
+                        $options['host'], $chunkDir, $filename
+                    );
+
+                    $this->logger->info(sprintf('Generating url: %s', $url));
+
+                    $urls[$contentTypeId][$lang][] = $url;
+                }
+            }
+        }
+
+        return $urls;
+    }
+
+    /**
+     * Generating export file.
+     *
+     * @param $content
+     * @param $chunkPath
+     * @param array $options
+     */
+    private function generateFile($content, $chunkPath, array $options)
+    {
+        $generator = $this->getGeneratorService($options);
+
+        $data = new ContentDataValue($content, $options);
+
+        $generator->reset();
+        $generator->startDocument($data);
+
+        $this->contentListElementGenerator->generateElement($generator, $content);
+
+        $this->fileSystemHelper->save($chunkPath, $generator->endDocument($data));
+
+        $this->logger->info(sprintf('Generating file: %s', $chunkPath));
+    }
+
+    /**
+     * Returns output generator service.
+     *
+     * @param array $options
+     *
+     * @return \eZ\Publish\Core\REST\Common\Output\Generator
+     */
+    private function getGeneratorService(array $options)
+    {
+        $generatorService = sprintf('ezpublish_rest.output.generator.%s', $options['requestContentType']);
```

Comment:
Using the container that way is really considered bad practice. It could be done with a factory, or by injecting all the writers where they're needed, and picking the right one on runtime.